astropy/io/fits/src/compressionmodule.c

```diff
@@ -334,14 +350,15 @@ int get_header_double(PyObject* header, const char* keyword, double* val,
 
 
 int get_header_float(PyObject* header, const char* keyword, float* val,
-                     float def) {
+                     float def, HeaderGetFlags flags) {
     double tmp;
-    int ret = get_header_double(header, keyword, &tmp, def);
+    int ret = get_header_double(header, keyword, &tmp, def, flags);
     if (ret == 0) {
         if (tmp >= FLT_MIN || tmp <= FLT_MAX) { /* don't care about infs! */
```

Comment:
@drdavella I ran this locally and this always succeeded even if I put in intended-to-fail values like `1e300`. The result is just `inf`. I'm not really sure about the coercion rules but it seems like I'm missing something.

Reply:
Can you add a unit test that reproduces the failure case? I have a few ideas but it would be helpful if I could play around with it locally.

Reply:
Actually, this condition is wrong. Sorry I missed this before. But shouldn't it be `&&` instead of `||`?

Reply:
Also, I think the the comparison actually needs to be this:
```c
if (fabs(tmp) >= FLT_MIN && fabs(tmp) <= FLT_MAX) {
 ...
```

This is because `FLT_MIN` is defined as the following:
>The value of this macro is the minimum normalized positive floating point number that is representable in type float. It is supposed to be no more than 1E-37.

So the smallest actual value that can be represented by a float is actually `-FLT_MAX`, but we also want to make sure we test for the smallest representable magnitude using `FLT_MIN`.

Reply:
Oups ... 😓